### PR TITLE
Fix: vela cluster join reports 'resource name may not be empty' error

### LINKED ISSUE #4994: `vela cluster join` reports "resource name may not be empty" error
**Describe the bug**

`vela cluster join vela cluster join .kube/member2config` reports "resource name may not be empty" error, it works well in previous versions.
image
image

**To Reproduce**

**Expected behavior**

**Screenshots**

**KubeVela Version**
1.6.0

**Cluster information**

1.21.5

**Additional context**

### PR DESCRIPTION
**Pull request by nuclearwu**
Signed-off-by: wuzhongjian <wuzhongjian_yewu@cmss.chinamobile.com>

### Description of your changes

Fixes #4994 

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/multicluster/cluster_management.go [hunk 1]
```diff
@@ -410,7 +410,7 @@ func JoinClusterByKubeConfig(ctx context.Context, cli client.Client, kubeconfigP
 		}
 	}
 	if cfg, ok := ctx.Value(KubeConfigContext).(*rest.Config); ok {
-		if err = SetClusterVersionInfo(ctx, cfg, clusterName); err != nil {
+		if err = SetClusterVersionInfo(ctx, cfg, clusterConfig.ClusterName); err != nil {
```
Comment (wonderflow) [inline pkg/multicluster/cluster_management.go:413]: clusterConfig.ClutserName usually is empty

Comment (nuclearwu) [reply pkg/multicluster/cluster_management.go:413]: no, please see the picture below 
image

Comment (nuclearwu) [reply pkg/multicluster/cluster_management.go:413]: In theory, as long as the kubeconfig read is no problem, the cluster name in the cluster configuration cannot be empty.

Comment (wonderflow) [reply pkg/multicluster/cluster_management.go:413]: ping @Somefive , please review

**Comment by Somefive** [comment]
This PR lgtm. I'm a bit curious that how #4994 happens? Could you explain it a bit more?

**Comment by nuclearwu** [comment]
> This PR lgtm. I'm a bit curious that how #4994 happens? Could you explain it a bit more?

This problem should be caused by #4890. When setting the cluster version information, the input parameter clusterName is obtained directly from cmd. Therefore, when the --name parameter is not specified when the cluster is managed by `vela cluster join ./kube/config`, the input parameter clusterName="" will lead to error occurs.

**Review by Somefive — APPROVED** [review]
@nuclearwu I see. You are right. Great catch! Thanks for looking into it :)

File: pkg/multicluster/cluster_management.go (hunk 1 above)
Comment (nuclearwu) [reply pkg/multicluster/cluster_management.go:413]: /cc @wonderflow

**Comment by wonderflow** [comment]
great job, thanks!

**Pull request merged**